### PR TITLE
Add max wait count for transfer leader operator.

### PR DESCRIPTION
**Pull request by qiuyesuifeng**

### CODE REVIEW

**Comment by siddontang** [comment]
why not handling directly in LeaderTransfer? 
I think it breaks the semantics for Once.

**Comment by qiuyesuifeng** [comment]
@siddontang Agreed, PTAL.

File: server/operator.go [hunk 1]
```diff
@@ -176,6 +176,8 @@ func (co *ChangePeerOperator) Do(region *metapb.Region, leader *metapb.Peer) (bo
 
 // TransferLeaderOperator is used to do leader transfer.
 type TransferLeaderOperator struct {
+	mustSuc bool
```
Comment (ngaut) [inline server/operator.go]: s/mustSuc/mustSuccess/

**Comment by siddontang** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Comment by qiuyesuifeng** [comment]
@shenli  @siddontang 
After the discuss, we all think using hb count is better, so i update this check.
PTAL

File: server/operator.go [hunk 2]
```diff
@@ -176,16 +182,19 @@ func (co *ChangePeerOperator) Do(region *metapb.Region, leader *metapb.Peer) (bo
...
 }
 
-func newTransferLeaderOperator(oldLeader, newLeader *metapb.Peer) *TransferLeaderOperator {
+func newTransferLeaderOperator(oldLeader, newLeader *metapb.Peer, waitCount int) *TransferLeaderOperator {
 	return &TransferLeaderOperator{
-		oldLeader: oldLeader,
-		newLeader: newLeader,
+		oldLeader:    oldLeader,
+		newLeader:    newLeader,
+		count:        0,
```
Comment (shenli) [inline server/operator.go]: The default value of count is already zero.

Comment (qiuyesuifeng) [inline server/operator.go]: Yes, it is a variable like index, we should update it in Do function.

File: server/operator.go [hunk 3]
```diff
@@ -220,16 +229,21 @@ func (lto *TransferLeaderOperator) Do(region *metapb.Region, leader *metapb.Peer
 		return true, nil, nil
 	}
 
-	// If lto.mustSuccess is true, then lto.check should always be true.
-	if lto.mustSuccess {
-		return false, nil, errors.Errorf("transfer leader operator called twice - %v", lto)
+	// If lto.count is greater than 0, then we should check whether it exceeds the lto.maxWaitCount.
+	if lto.count > 0 {
```
Comment (shenli) [inline server/operator.go]: If some one set maxWaitCount to zero, you will get infinite loop. So it is better to add a check in newTransferLeaderOperator or here.

Comment (qiuyesuifeng) [inline server/operator.go]: Yep, later we will move it as config flag.
Then we can check it.
Now we have set it directly, so it will never be zero.

**Comment by shenli** [comment]
LGTM

File: server/operator.go (hunk 2 above)
Comment (siddontang) [inline server/operator.go]: `count` and `maxWaitCount` confuse me. 

you can initialize count as waitCount and then use `lto.count < 0 lt.count--` to check.

Comment (qiuyesuifeng) [inline server/operator.go]: Seems not, we must store the origin `maxWaitCount` to make sure the first time we return the transfer leader response. 
If we only use `count`, then we cannot distinct it.

Comment (siddontang) [inline server/operator.go]: Got it.

**Pull request merged**